test/integration/conftest.py

```diff
@@ -188,21 +192,71 @@ def test_case(api_database):
 
 
 @pytest.fixture(scope="module")
-def api_database_module(api_isolation_module: ApiTestEnvironment) -> ApiContextProvider:
```

Comment:
Replaced `api_default_database_module`.
Reason: If tests need a database for the whole module, it's unlikely they need customization (e.g. `additional_parameters`)
If they need customization, they should create their own fixture in the module.

Reply:
Then we should make _build_api_context_provider_with_test_container public and move it out of conftest

Reply:
Done